### ISSUE TITLE
Getting up to speed with this codebase

### ISSUE THREAD

**Issue by vitosamson** [francose/HAB#1]
Hey @francose,

I'm getting involved with the HAB project and need to get up to speed on this codebase. Docs and comments are sparse so I'm having a hard time piecing it all together.

Some questions:

1. The `HAB.ino` file relies on a bunch of header files, like `SPI.h`, `SD.h`, etc. Where can I get those libraries?
2. What does the build process entail? Is there anything special I need to do to cross-compile this for the arduino?
3. Is `StemHab.cpp` used at all? I see `StemHab.h` included in `HAB.ino`, but it doesn't look like the functions are used.
4. How were you testing this code previously?

Are you still at BCC at all, and if so, would you be able to meet up at some point to go over this?

Thanks,
Vito

**Comment by francose** [comment]
* The `HAB.ino` file relies on a bunch of header files, like `SPI.h`, `SD.h`, etc. Where can I get those libraries?

**you need those headers for the _datalogging_  SD card .
https://www.arduino.cc/en/Reference/SD
https://www.arduino.cc/en/Tutorial/Datalogger**


* What does the build process entail? Is there anything special I need to do to cross-compile this for the arduino?

**installation:
_you can visit adriuno 's website  => https://www.arduino.cc/en/guide/macOSX#toc1
you need to install all the required packages for that specific board so you can compile the code._**


* Is `StemHab.cpp` used at all? I see `StemHab.h` included in `HAB.ino`, but it doesn't look like the functions are used.

**As i mention earlier, i didnt used them. they are coming from the previous team.** 

* How were you testing this code previously?

**Yes we have tested that code with Vlad, it was running. The only non functioning part on the code is pressure sensor and motor functions.**

**Comment by vitosamson** [comment]
Thanks, I was able to get it setup and compiled in the Arduino app.

I'm trying to make sense of the parsing of the serial data [here](https://github.com/francose/HAB/blob/master/HAB.ino#L164).

Is there any documentation for the output of the sensors?

Do the GPS, temperature, and pressure sensors all go through the OneWire library? 

How do you know what order the values will be sent over the serial interface, and what format they're in? I couldn't find any info for this in the PDF docs in this repo.

**Comment by francose** [comment]
For the documentation i think it should be better if you use the manufacturers docs from their website. 
You do not need to use OneWire you can use some other libraries or write your one just we did for the GPS module. We had some old documentations in https://mega.nz/ however i dont have the credentials tho. so you can definitely look up for more docs.

for the gps when i parse it, it catches everything by character. Then it basically gather all the data and write into a csv file.  For the rest of the libraries you need to check their docs.

**Comment by vitosamson** [comment]
Got it.

So the GPS data is being parsed on the fly and then only certain values are being written out to disk. Is there a reason the entire GPGGA sentence isn't being written out while the balloon is in the air and then parsed once the payload has been retrieved?

**Comment by francose** [comment]
It reads from the GPS module straight once it spits out the data. We needed the data right away as soon as the terminal starts.

**Comment by vitosamson** [comment]
Right, but when it writes the GPS data to disk, it only writes certain pieces of data [here](https://github.com/francose/HAB/blob/master/HAB.ino#L211). I'm just wondering if there was a reason to not just write out `newLine.full` and parse it after retrieval. 

Aside from grabbing the altitude to determine if the door needs to be opened or closed, I'm not sure what the reason is for parsing the GPS data into the `singleLineParsed` struct and then writing those struct fields out to the file.

**Comment by francose** [comment]
you can definitely modify however you want.
